bin/build.sh

```diff
@@ -418,52 +360,34 @@ if [ ${INSTALL_CVC4} -eq 1 ] ; then
   else
     puts "CVC4 already installed"
   fi
+  echo export PATH=\"${CVC4_DIR}:\$PATH\" >> ${SMACKENV}
 fi
 
-if [ ${BUILD_BOOGIE} -eq 1 ] ; then
-  if ! upToDate $BOOGIE_DIR $BOOGIE_COMMIT ; then
-    puts "Building Boogie"
-    if [ ! -d "$BOOGIE_DIR/.git" ] ; then
-      git clone https://github.com/boogie-org/boogie.git ${BOOGIE_DIR}
-    fi
-    cd ${BOOGIE_DIR}
-    git reset --hard ${BOOGIE_COMMIT}
-    cd ${BOOGIE_DIR}/Source
-    ${WGET} https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
-    mono ./nuget.exe restore Boogie.sln
-    rm -rf /tmp/nuget/
-    msbuild Boogie.sln /p:Configuration=Release
-    ln -sf ${Z3_DIR}/bin/z3 ${BOOGIE_DIR}/Binaries/z3.exe
-    ln -sf ${CVC4_DIR}/cvc4 ${BOOGIE_DIR}/Binaries/cvc4.exe
-    puts "Built Boogie"
+
+if [ ${INSTALL_BOOGIE} -eq 1 ] ; then
+  if [ ! -d "$BOOGIE_DIR" ] ; then
```

Comment:
You could simplify a bit here:
* Dotnet tools are content with sharing tool path, so you could replace `BOOGIE_DIR` and `CORRAL_DIR` with `DOTNET_TOOL_PATH` — or even just use the default global path.
* Detecting and reporting on whether Boogie/Corral is “already installed” seems redundant with `dotnet tool install`, which will do that anyways.

Reply:
Good catch! Done.